usaspending_api/broker/management/commands/load_fpds_from_broker.py

```diff
@@ -0,0 +1,139 @@
+from django.core.management.base import BaseCommand
+import logging
+import re
+import psycopg2
+from datetime import datetime, timezone, timedelta
+
+from usaspending_api.data_load.fpds_loader import run_fpds_load, destroy_orphans, load_chunk
+from usaspending_api.common.retrieve_file_from_uri import RetrieveFileFromUri
+from usaspending_api.common.helpers.date_helper import datetime_command_line_argument_type
+from usaspending_api.common.helpers.sql_helpers import get_broker_dsn_string
+from usaspending_api.common.helpers.etl_helpers import update_c_to_d_linkages
+from usaspending_api.etl.award_helpers import update_awards, update_contract_awards, update_award_categories
+from usaspending_api.broker.helpers.last_load_date import get_last_load_date, update_last_load_date
+
+logger = logging.getLogger("console")
+
+BROKER_CONNECTION_STRING = get_broker_dsn_string()
+
+CHUNK_SIZE = 5000
+
+ALL_FPDS_QUERY = "SELECT * FROM detached_award_procurement"
+
+
+class Command(BaseCommand):
+    help = "Sync USAspending DB FPDS data using Broker for new or modified records and S3 for deleted IDs"
+
+    modified_award_ids = []
+
+    @staticmethod
+    def get_cursor_for_date_query(connection, date):
+        db_cursor = connection.cursor("fpds_load", cursor_factory=psycopg2.extras.DictCursor)
+        db_query = ALL_FPDS_QUERY
+        if date:
+            db_query += " WHERE updated_at >= %s;"
+            db_args = [date]
+            db_cursor.execute(db_query, db_args)
+        else:
+            db_cursor.execute(db_query)
+        return db_cursor
+
+    def load_fpds_from_date(self, date):
+        if date is None:
+            logger.info("fetching all fpds transactions...")
+        else:
+            logger.info("fetching fpds transactions since {}...".format(str(date)))
+        with psycopg2.connect(dsn=BROKER_CONNECTION_STRING) as connection:
+            cursor = self.get_cursor_for_date_query(connection, date)
```

Comment:
I'm not sure how to get info on the first, since we are trying to not get everything at once, and I'm not sure what "insights" you're looking for with the second point.

Reply:
At a minimum the total number of records needs to be reported at the end.

It is possible to obtain the transaction count at the beginning of the script before the named cursor is run to obtain the records. 

> since we are trying to not get everything at once
Getting a count is simply an integer, no memory load required.

It would add time, but it would benefit the people watching the script run (specifically, me at the moment) and can be used for rough script completion estimates.

Reply:
changes have been made to `load_fpds_from_date`

Reply:
great, thanks